app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -919,65 +969,17 @@ export default Component.extend(ComposerUploadUppy, {
       });
     },
 
-    previewUpdated(preview) {
-      // cache jquery objects for functions still using jquery
-      const $preview = $(preview);
-
-      // Paint mentions
-      const unseenMentions = linkSeenMentions(preview, this.siteSettings);
-      if (unseenMentions.length) {
-        discourseDebounce(
-          this,
-          this._renderUnseenMentions,
-          preview,
-          unseenMentions,
-          450
-        );
-      }
+    previewUpdated(preview, unseenMentions, unseenHashtags) {
+      this._renderMentions(preview, unseenMentions);
+      this._renderHashtags(preview, unseenHashtags);
+      this._refreshOneboxes(preview);
+      this._expandShortUrls(preview);
 
-      this._warnMentionedGroups(preview);
-      this._warnCannotSeeMention(preview);
-
-      // Paint category, tag, and other data source hashtags
-      const hashtagContext = this.site.hashtag_configurations["topic-composer"];
-      if (linkSeenHashtagsInContext(hashtagContext, preview).length > 0) {
-        discourseDebounce(this, this._renderUnseenHashtags, preview, 450);
+      if (!this.siteSettings.enable_diffhtml_preview) {
+        this._decorateCookedElement(preview);
```

Comment:
This will call all the `api.decorateCookedElement` so we don't want to do them twice when morphing.